core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java

```diff
@@ -148,8 +153,26 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
     }
     dlpPositiveItemEntity.setType(TYPE);
     dlpPositiveItemEntity.setDetectionDate(Calendar.getInstance());
-    // to be updated with detected keyword
-    dlpPositiveItemEntity.setKeywords(dlpKeywords);
+    dlpPositiveItemEntity.setKeywords(getDetectedKeywords(searchResults));
     dlpPositiveItemService.addDlpPositiveItem(dlpPositiveItemEntity);
   }
+
+  private String getDetectedKeywords(Collection<SearchResult> searchResults) {
+    List<String> excerptsList = new ArrayList<>();
+    for (SearchResult searchResult : searchResults) {
+      Map<String, List<String>> excerpts = searchResult.getExcerpts();
+      for (List<String> excerptValue : excerpts.values()) {
+        if (excerptValue != null && !excerptValue.isEmpty()) {
+          Pattern pattern = Pattern.compile("<em>(.*?)</em>", Pattern.DOTALL);
+          Matcher matcher = pattern.matcher(excerptValue.get(0).toString());
+          while (matcher.find()) {
+            if (!excerptsList.contains(matcher.group(1))) {
+              excerptsList.add(matcher.group(1));
+            }
+          }
+        }
+      }
+    }
+    return StringUtils.join(excerptsList, ", ");
```

Comment:
```suggestion
    return excerptsList.stream().collect(Collectors.joining(","));
```
I prefer to use join directly from java instead of using external lib. 
In addition you can remove the StringUtils import